### PR TITLE
doc (#375): Document complex use cases of Env.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -246,7 +246,44 @@ public class Main {
 }
 ```
 
-Additional options are provided for adding environment variables from fields, config maps and secrets.    
+Additional options are provided for adding environment variables from fields, config maps and secrets. 
+
+#### Adding environmnet variables from ConfigMap
```
Comment (geoand) [inline readme.md]: `environmnet` -> `environment`

File: readme.md [hunk 2]
```diff
@@ -246,7 +246,44 @@ public class Main {
...
+
+@KubernetesApplication(envVars = @Env(name = "key1", configmap="my-config", value = "key1"))
+public class Main {
+
+  public static void main(String[] args) {
+    //Your code goes here
+  }
+}
+```
+#### Adding environmnet variables from Secrets
```
Comment (geoand) [inline readme.md]: `environmnet` -> `environment`

File: readme.md [hunk 3]
```diff
@@ -246,7 +246,44 @@ public class Main {
 }
 ```
 
-Additional options are provided for adding environment variables from fields, config maps and secrets.    
+Additional options are provided for adding environment variables from fields, config maps and secrets. 
+
+#### Adding environmnet variables from ConfigMap
+
+To add an environment variable that points to a ConfigMap property, you need to specify the configmap using the `configmap` property in the annotation.
```
Comment (aureamunoz) [inline readme.md]: I would change  `in the annotation` --> `in the @Env annotation` (My first reflection has been to look for an `@ConfigMap` annotation in the code to verify this field)

File: readme.md [hunk 4]
```diff
@@ -246,7 +246,44 @@ public class Main {
...
+public class Main {
+
+  public static void main(String[] args) {
+    //Your code goes here
+  }
+}
+```
+#### Adding environmnet variables from Secrets
+
+To add an environment variable that points to a Secret property, you need to specify the configmap using the `secret` property in the annotation.
```
Comment (aureamunoz) [inline readme.md]: Same here, add `@Env`

File: readme.md (hunk 1 above)
Comment (iocanel) [reply readme.md]: fixed.

File: readme.md (hunk 2 above)
Comment (iocanel) [reply readme.md]: fixed

File: readme.md (hunk 3 above)
Comment (iocanel) [reply readme.md]: fixed.

File: readme.md (hunk 4 above)
Comment (iocanel) [reply readme.md]: fixed.

**Pull request merged**